src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+				common.CreateDockerClient,
+				configClosure,
+				&brokers,
+			)
+			tlsClosure := common.BuildKafkaTLSConfig(fs, &enableTLS, &certFile, &keyFile, &truststoreFile, configClosure)
+			kAuthClosure := common.KafkaAuthConfig(&user, &password, &mechanism, configClosure)
+			admin := common.CreateAdmin(brokersClosure, configClosure, tlsClosure, kAuthClosure)
+
+			adm, err := admin()
+			if err != nil {
+				log.Error("Couldn't initialize API admin")
+				return err
+			}
+			defer adm.Close()
+
+			return executeSnapshot(fs, conf, adm, logsSince, logsUntil, timeout)
+		},
+	}
+	command.Flags().DurationVar(
+		&timeout,
+		"timeout",
+		10*time.Second,
+		"The maximum amount of time to wait for child commands (ss, ifconfig, "+
+			"dig, etc.) to be executed. "+
+			"The value passed is a sequence of decimal numbers, each with optional "+
+			"fraction and a unit suffix, such as '300ms', '1.5s' or '2h45m'. "+
+			"Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'",
+	)
+	command.Flags().StringVar(
+		&logsSince,
+		"logs-since",
+		"",
+		fmt.Sprintf(`Include log entries on or newer than the specified date. %s`, timeHelpText),
+	)
+	command.Flags().StringVar(
+		&logsUntil,
+		"logs-until",
+		"",
+		fmt.Sprintf(`Include log entries on or older than the specified date. %s`, timeHelpText),
+	)
+
+	common.AddKafkaFlags(
+		command,
+		&configFile,
+		&user,
+		&password,
+		&mechanism,
+		&enableTLS,
+		&certFile,
+		&keyFile,
+		&truststoreFile,
+		&brokers,
+	)
+	return command
+}
+
+func executeSnapshot(
+	fs afero.Fs,
+	conf *config.Config,
+	admin sarama.ClusterAdmin,
+	logsSince, logsUntil string,
+	timeout time.Duration,
+) error {
+	mode := os.FileMode(0755)
+	timestamp := time.Now().Unix()
+	filename := fmt.Sprintf("%d-bundle.zip", timestamp)
+	f, err := fs.OpenFile(
+		filename,
+		os.O_CREATE|os.O_WRONLY,
+		mode,
+	)
+	if err != nil {
+		return fmt.Errorf("couldn't create bundle file: %w", err)
+	}
+	defer f.Close()
+
+	grp := multierror.Group{}
+
+	m := &sync.Mutex{}
+	w := zip.NewWriter(f)
+	defer w.Close()
+
+	ps := &stepParams{fs: fs, m: m, w: w, timeout: timeout}
+
+	steps := []step{
+		saveKafkaMetadata(ps, admin),
+		saveLogs(ps, logsSince, logsUntil),
+		saveSyslog(ps),
+		saveConfig(ps, conf),
+		saveResourceUsageData(ps, conf),
+		saveNTPDrift(ps),
+		saveDNSData(ps),
+		saveCPUInfo(ps),
+		saveIfconfig(ps),
+		saveVmstat(ps),
+		saveInterrupts(ps),
+	}
+	// each step will send the file it produced through producedFiles, which
+	// will be used to build a manifest.
+	ps.producedFiles = make(chan string, len(steps))
+
+	for _, s := range steps {
+		grp.Go(s)
+	}
+
+	errs := grp.Wait()
+	if errs != nil {
+		err := writeFileToZip(ps, "errors.txt", []byte(errs.Error()))
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		log.Info(errs.Error())
+	}
+
+	close(ps.producedFiles)
+
+	log.Infof("Debug bundle saved to '%s'", filename)
+
+	manifestFilename := fmt.Sprintf("%d-bundle-manifest.txt", timestamp)
```

Comment:
Is the manifest important? I think it's easier for users to just send us one file rather than two, and the manifest we receive may not reflect reality -- I usually prefer not to have a separate file list. If we want something like this we may want to create a tarball, which allows for file inspection without decompressing everything.

Reply:
@esteban especifically asked for this.

Reply:
I've pinged @esteban a bit and I think we need to define the manifest file. A simple list of files in the zip itself is not useful (because we can look at the list of files in a zip without decompressing it). We'd want the manifest to have more expanded information about the commands run,

"""
a collection of files and commands run by the bundle command with finer detail about the results of the execution of each command, e.g. did we get stderr? what was the exit code? if we run a command why it failed passing eth0 instead of en0? and so on
"""

This could be either a csv or json (probably json, since it's easier to add to).

We currently don't have a definition of a manifest file, and I don't think we want to try to spec out a manifest right now when we have so few bundles being generated and aren't running _that_ much. We can spec out the manifest file later, but I think we should drop the manifest from this PR itself. When we do have an expanded manifest later, the json type manifest can be bundled in the zip itself as well.

Reply:
Let me check with him to see if we can drop it or if we need to define the manifest well and include it in this PR.

Reply:
Manifest: removed.